packages/integration_test/ios/Classes/FLTIntegrationTestCase.m

```diff
@@ -0,0 +1,76 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// XCTest is weakly linked.
+#if __has_include(<XCTest/XCTest.h>)
+
+#import "FLTIntegrationTestCase.h"
+
+#import "IntegrationTestIosTest.h"
+#import "IntegrationTestPlugin.h"
+
+@import ObjectiveC.runtime;
+@import XCTest;
+
+@implementation FLTIntegrationTestCase
+
++ (NSArray<NSInvocation *> *)testInvocations {
+  if (self == [FLTIntegrationTestCase class]) {
+    // Do not add any tests for this base class.
+    return @[];
+  }
+  IntegrationTestIosTest *integrationTestIosTest = [IntegrationTestIosTest new];
+  NSMutableArray<NSInvocation *> *testInvocations = [NSMutableArray new];
+  [integrationTestIosTest testIntegrationTestWithResults:^(NSString *testName, BOOL success, NSString *_Nullable failureMessage) {
+    // For every Flutter dart test, dynamically generate an Objective-C method mirroring the test results
+    // so it is reported as a native XCTest run result.
```

Comment:
This is slick! Out of curiosity, do we know if it's guaranteed to work? It seems a bit modifying-an-iterator-while-iterating, which makes me wonder if it has defined behavior.

Reply:
That's actually the purpose of `+ (NSArray<NSInvocation *> *)testInvocations`--a hook for dynamically changing the test methods to use for a test case.  https://stackoverflow.com/a/55204082 was a helpful reference.